### ISSUE TITLE
outage due to `LD_PRELOAD` changes in Colab

### ISSUE THREAD

**Issue by francesco-ballarin** [fem-on-colab/fem-on-colab#19]
It seems that `LD_PRELOAD` is not getting exported anymore on actual Colab runtime instances.

This is affecting e.g. FEniCS I/O, but would more generally require a change of policy in terms of the statically linking to `libstdc++`.

A fix is in progress, and will require recompiling the entire stack.

**Comment by francesco-ballarin** [comment]
`gmsh` is probably the package most noticeably affected.

**Comment by francesco-ballarin** [comment]
I asked at https://github.com/googlecolab/colabtools/issues/3397 about these changes

**Comment by shlee-git** [comment]
yes, also in my case, gmsh is broken!

**Comment by rubensamarojr** [comment]
The same here.
I'm trying to save the solution from fenics legacy in XDMF or PVD format, but none of them is producing a valid file for Paraview.

**Comment by niravshah241** [comment]
gmsh is broken for me as well.

**Comment by francesco-ballarin** [comment]
A fix has been released, which basically consists in replacing the system `libstdc++.so` in `/usr/lib` with the one which we install when building `gcc` 12. Unfortunately, a kernel restart must be forced in order for the replacement to have effect. To this end, the kernel is crashed on purpose: users are informed of what is happening via a text message that hopefully should be noticeable enough, and even directed to the issue in the google colab repo.

I regret having to force a kernel crash to fix this, but after a month of work this was the best solution I found. I'll go back to the former behavior if upstream will reconsider at some point the breaking change.
